src/parameterizations/vertical/MOM_geothermal.F90

```diff
@@ -304,6 +316,14 @@ subroutine geothermal(h, tv, dt, ea, eb, G, GV, CS, halo)
     enddo ; endif
   enddo ! j-loop
 
+! Calculate heat tendency due to addition and transfer of internal heat
+if (CS%id_internal_heat_tend_3d > 0) then
+  do k=1,nz ; do j=js,je ; do i=is,ie
+    work_3d(i,j,k) = GV%H_to_kg_m2 * tv%C_p * Idt * (h(i,j,k) * tv%T(i,j,k) - h_old(i,j,k) * temp_old(i,j,k))
+  enddo ; enddo ; enddo
+  call post_data(CS%id_internal_heat_tend_3d, work_3d, CS%diag, alt_h = h_old)
+endif
+
```

Comment:
This block also need to be align with code block above.

Reply:
Because you're already making the change, could you add a diagnostic for `h` also? I actually didn't realize that this routine could change thicknesses; which will be good to know if ever someone needs to reconstruct thickness budgets.